controller/controlplane/controller.go

```diff
@@ -468,7 +456,7 @@ func (r *Reconciler) patchStatus(ctx context.Context, logger logr.Logger, update
 	}
 
 	log.Debug(logger, "patching ControlPlane status", "status", updated.Status)
-	if err := r.Client.Status().Patch(ctx, updated, client.MergeFrom(current)); err != nil {
+	if err := r.Client.Status().Patch(ctx, updated, client.MergeFrom(current)); err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
```suggestion
	if err := r.Client.Status().Patch(ctx, updated, client.MergeFrom(current)); client.IgnoreNotFound(err) != nil {
```
from package `"sigs.k8s.io/controller-runtime/pkg/client"`

WDYT?